modules/metastorage/src/test/java/org/apache/ignite/internal/metastorage/server/AbstractCompactionKeyValueStorageTest.java

```diff
@@ -270,6 +274,58 @@ void testEntryOperationTimestampAfterCompaction() {
         assertNull(storage.get(BAR_KEY).timestamp());
     }
 
+    @Test
+    void testTimestampByRevisionAfterCompaction() {
+        storage.compact(1);
+
+        assertThrows(CompactedException.class, () -> storage.timestampByRevision(1));
+        assertDoesNotThrow(() -> storage.timestampByRevision(2));
+        assertDoesNotThrow(() -> storage.timestampByRevision(3));
+
+        storage.compact(2);
+
+        assertThrows(CompactedException.class, () -> storage.timestampByRevision(1));
+        assertThrows(CompactedException.class, () -> storage.timestampByRevision(2));
+        assertDoesNotThrow(() -> storage.timestampByRevision(3));
+
+        storage.compact(3);
+
+        assertThrows(CompactedException.class, () -> storage.timestampByRevision(1));
+        assertThrows(CompactedException.class, () -> storage.timestampByRevision(2));
+        assertThrows(CompactedException.class, () -> storage.timestampByRevision(3));
+    }
+
+    @Test
+    void testRevisionByTimestampAfterCompaction() {
+        HybridTimestamp timestamp0 = storage.timestampByRevision(1);
+        HybridTimestamp timestamp1 = storage.timestampByRevision(2);
+        HybridTimestamp timestamp2 = storage.timestampByRevision(3);
```

Comment:
```suggestion
        HybridTimestamp timestamp1 = storage.timestampByRevision(1);
        HybridTimestamp timestamp2 = storage.timestampByRevision(2);
        HybridTimestamp timestamp3 = storage.timestampByRevision(3);
```
Otherwise it looks weird: 0-based on the left and 1-based on the right

Reply:
Why not.